src/main/java/org/ibp/api/rest/study/StudyEntryObservationResource.java

```diff
@@ -0,0 +1,80 @@
+package org.ibp.api.rest.study;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.generationcp.middleware.service.api.dataset.StockPropertyData;
+import org.ibp.api.java.study.StudyEntryObservationService;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.annotation.Resource;
+import java.util.Arrays;
+
+@Api(value = "Study Entry Observation Services")
+@Controller
+public class StudyEntryObservationResource {
+
+	@Resource
+	private StudyEntryObservationService studyEntryObservationService;
+
+	@ApiOperation(value = "Add new observation for a given study entry", notes = "Add new observation for a given study entry")
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES')")
+	@RequestMapping(value = "/crops/{cropName}/programs/{programUUID}/studies/{studyId}/observations", method = RequestMethod.POST)
+	public ResponseEntity<Integer> createObservation(
+		@PathVariable final String cropName,
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId,
+		@RequestBody final StockPropertyData stockPropertyData) {
+		final Integer stockPropertyId = this.studyEntryObservationService.createObservation(studyId, stockPropertyData);
+		return new ResponseEntity<>(stockPropertyId, HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "Updates the given study entry observation", notes = "Updates the given study entry observation")
+	@PreAuthorize("hasAnyAuthority('ADMIN','STUDIES','MANAGE_STUDIES')")
+	@RequestMapping(value = "/crops/{cropName}/programs/{programUUID}/studies/{studyId}/observations", method = RequestMethod.PATCH)
```

Comment:
@lik-83 observationId should be in the URL and the only values that can be modified is the value, assuming that cvalueId is calculated internally when variable is categorical. It was creating some confussion while testing and I did not noticed before. If it will create many changes in the frontend now, please create a ticket and we improve it in a new branch

Reply:
Sure! This is the ticket: https://ibplatform.atlassian.net/browse/IBP-5795